### PR TITLE
Add testing utilities to documents

### LINKED ISSUE #2880: Register a testing utility to the document
It is recommended for developers to use [`chainer.testing.get_trainer_with_mock_updater`](https://github.com/chainer/chainer/blob/master/chainer/testing/training.py#L8) to test trainers and related modules. But it is not registered to the document.

### PR DESCRIPTION
**Pull request by delta2323**
This PR adds several utility functions that are missing to the document.
This PR fixes #2880

### CODE REVIEW

File: docs/source/reference/check.rst [hunk 1]
```diff
@@ -55,3 +56,54 @@ Chainer provides some utilities for testing its functions.
    :nosignatures:
 
    chainer.testing.unary_math_function_unittest
+   chainer.testing.save_and_load
+   chainer.testing.save_and_load_hdf5
+   chainer.testing.save_and_load_npz
+   chainer.testing.get_trainer_with_mock_updater
+
```
Comment (kmaehashi) [inline docs/source/reference/check.rst]: I think these are mainly for Trainer extensions, so it shouldn't be under `Function testing utilities` catgory.
How about adding a new category for these functions?

File: docs/source/reference/index.rst [hunk 2]
```diff
@@ -22,4 +22,4 @@ Reference
    datasets
    iterators
    caffe
-   graph
+   graph
```
Comment (kmaehashi) [inline docs/source/reference/index.rst]: This change seems irrelevant to this PR.

File: docs/source/reference/check.rst [hunk 3]
```diff
@@ -55,3 +56,54 @@ Chainer provides some utilities for testing its functions.
...
+
+Repeat decorators
+-----------------
+
+These decorators have a decorated test run multiple times
```
Comment (kmaehashi) [inline docs/source/reference/check.rst]: -`These decorators have a decorated test run multiple times in a single invocation.`
+`These decorators can be used to decorate test to run mulitple times in a single invocation.`

File: docs/source/reference/check.rst [hunk 4]
```diff
@@ -55,3 +56,54 @@ Chainer provides some utilities for testing its functions.
...
+in a single invocation. Criteria of passing / failing
+of the test changes according to the type of decorators.
+See the document of each decorator in detail.
```
Comment (kmaehashi) [inline docs/source/reference/check.rst]: `in detail` -> `for details`?

**Comment by kmaehashi** [comment]
@delta2323 ping? (I can takeover if you don't have time!)

**Comment by delta2323** [comment]
@kmaehashi Sorry for late reply, Could you take over the PR?
